lib/components/form/call-taker/date-time-picker.tsx

```diff
@@ -0,0 +1,235 @@
+      const resolvedDateTime = format(
+        parse(timeInput, matchedTimeFormat, new Date()),
+        'HH:mm:ss'
+      )
+      return toDate(`${date}T${resolvedDateTime}`)
+    }
+    return ''
+  }
+
+  const dateTime = parseInputAsTime(time, date)
+
+  // Handler for setting the query parameters
+  useEffect(() => {
+    if (safeFormat(dateTime, OTP_API_DATE_FORMAT, {}) !== '' && setQueryParam) {
+      setQueryParam({
+        date: safeFormat(dateTime, OTP_API_DATE_FORMAT, {
+          timeZone: homeTimezone
+        }),
+        departArrive,
+        time: safeFormat(dateTime, OTP_API_TIME_FORMAT, {
+          timeZone: homeTimezone
+        })
+      })
+    }
+  }, [dateTime, departArrive, homeTimezone, setQueryParam])
+
+  // Handler for updating the time and date fields when NOW is selected
+  useEffect(() => {
+    if (departArrive === 'NOW') {
+      setTime(getCurrentTime(homeTimezone))
+      setDate(getCurrentDate(homeTimezone))
+    }
+  }, [departArrive, setTime, setDate, homeTimezone])
+
+  const unsetNow = () => {
+    if (departArrive === 'NOW') setDepartArrive('DEPART')
+  }
+
+  return (
+    <>
+      <select
+        onBlur={(e) => setDepartArrive(e.target.value)}
+        onChange={(e) => setDepartArrive(e.target.value)}
+        onKeyDown={onKeyDown}
+        value={departArrive}
+      >
+        {getDepartureOptions(intl).map(({ text, value }) => (
+          <option key={value} value={value}>
+            {text}
+          </option>
+        ))}
+      </select>
+      <OverlayTrigger
+        overlay={
+          <Tooltip id="time-tooltip">
+            {safeFormat(dateTime, timeFormat, {
+              timeZone: homeTimezone
+            })}
+          </Tooltip>
+        }
+        placement="bottom"
+        trigger={['focus', 'hover']}
+      >
+        <input
+          className="datetime-slim"
+          onChange={(e) => {
+            setTime(e.target.value)
+            unsetNow()
+          }}
+          onFocus={(e) => e.target.select()}
+          onKeyDown={onKeyDown}
+          style={{
+            fontSize: 'inherit',
+            lineHeight: '.8em',
+            marginLeft: '3px',
+            padding: '0px',
+            width: '50px'
+          }}
+          // Don't use intl.formatTime, so that users can enter time in 12hr or 24hr format at their leisure.
+          value={
+            time && time?.length > 1
+              ? time || format(dateTime, 'H:mm', { timeZone: homeTimezone })
+              : time
+          }
+        />
+      </OverlayTrigger>
+      <input
+        className="datetime-slim"
+        onChange={(e) => {
+          setDate(e.target.value)
+          unsetNow()
+        }}
+        onKeyDown={onKeyDown}
+        style={{
+          fontSize: '14px',
+          lineHeight: '1em',
+          outline: 'none',
+          width: '109px'
+        }}
+        type="date"
+        value={safeFormat(dateTime, OTP_API_DATE_FORMAT, {
+          timeZone: homeTimezone
```

Comment:
Very small issue but this line causes the date to zero out if the time input is invalid: 
https://user-images.githubusercontent.com/115499534/208951708-5c17ac68-6be7-438e-88ea-4fd6faccb800.mov

Reply:
What's the expected behavior here? The trip can't be planned if the date is wrong. Maybe it should be grayed out let me try that

Reply:
I think it would be enough just for the date field to stay consistent, even if there's an error in the time field! But honestly I don't think it's too distracting as is!